src/validators/check_id.rs

```diff
@@ -1,26 +1,10 @@
 use crate::issues::*;
 
 pub fn validate(gtfs: &gtfs_structures::Gtfs) -> Vec<Issue> {
-    let r = gtfs
-        .routes
-        .values()
-        .filter(missing_id)
-        .map(make_missing_id_issue);
-    let t = gtfs
-        .trips
-        .values()
-        .filter(missing_id)
-        .map(make_missing_id_issue);
-    let c = gtfs
-        .calendar
-        .values()
-        .filter(missing_id)
-        .map(make_missing_id_issue);
-    let st = gtfs
-        .stops
-        .values()
-        .filter(missing_id)
-        .map(make_missing_id_issue);
+    let r = gtfs.routes.values().flat_map(valid_id);
```

Comment:
don't you want a `filter_map` here?

Reply:
indeed, now I’m a bit confused. Does `filter_map` bring anything compared to `flat_map`? (excepted the semantics)

Reply:
While reading it took me a while to understand, but I think it's just a side effect because `Option` implements `IntoIter` so it works the same (trait magic 🪄 🤯 ), but the semantic is clearer with filter_map😜